src/libs/collect.c

```diff
@@ -2106,7 +2106,7 @@ static void row_activated_with_event(GtkTreeView *view, GtkTreePath *path, GtkTr
          * hierarchy. */
         else if(event->state & GDK_SHIFT_MASK)
         {
-          gchar *n_text = g_strconcat(text, "%", NULL);
+          gchar *n_text = g_strconcat(text, "*", NULL);
```

Comment:
Ok, let's try to improve that. We have:
- click (no wildchar)
- ctrl-click  `|%` at end
- shift-click `%` at end

You changed the shift-click to use `*` but the tooltip is now wrong.

I propose to keep `%` at end (which is uniform with all others collection kind) and check if we end with `%` and not with `|%` in the code above. Should work too, and will not introduce a new wildchar.

Reply:
It was on purpose to introduce a new character because I don't want to change the current wildcard-behaviour.

If a user wants to get all tags starting with 'birthday' he can now enter birthday% and gets tags like 'birthday mary' and 'birthday john'.

This would not work anymore if the new hierarchy-logic uses the same character % as trigger.

So the tooltip is not wrong, because % still works for the birthday% usecase.

Maybe it should be another character instead of * to make it clear that this is a special-function and no wildcard.

Any suggestions?

Reply:
I was thinking about that indeed ! So what I propose is to just update the tooltip for this new wildchar and explain its usage.

Reply:
So we are talking about the tooltip "type your query, use `%' as wildcard", right?

I would prefer to keep this as is, because:
- the % wildcard works for all collections, the * function is limited to tags
- the wildcard % is obvious to be entered in the textbox, but the function of the asterisk is not trivial
- so I would not enter * manually in the textbox but only use the shift-click, so a longer tooltip could be confusing

Instead it could be better to promote the ctrl-click and shift-click possibility
- tooltip when hovering a tag in the list
- describe it in the user manual

Reply:
You can special case the tooltip for the tag and geotag (as this is where you have changed the semantic). See, there is already different tooltip for different collections.

Reply:
yes, that's true, I realized that after clicking 'comment'...

Nevertheless I would prefer to explain the shift-click functionality when hovering the tag in the list, because this would be much easier to understand compared to manually adding an asterisk in the textbox.

But if it's better to just change the existing tooltip for tags I will do it of course

Reply:
Not sure it is easy to add a tooltip to all entries into the treeview! At least, in my view, we need to add the asterisk into the current tooltip as this is the way we current document the use of the search box. So better do it all at the same place, the tricky part is to come with a short and clear description :)